spec/features/bulkrax_export_spec.rb

```diff
@@ -0,0 +1,126 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe 'Bulkrax export', clean: true, perform_enqueued: true do
+  let(:user) { create(:user, email: 'test@example.com') }
+  # let! is needed below to ensure that this user is created for file attachment because this is the depositor in the CSV fixtures
+  let(:depositor) { create(:user, email: 'batchuser@example.com') }
+  let(:exporter) do
+    create(:bulkrax_exporter_worktype,
+           user: user,
+           field_mapping: Bulkrax.field_mappings["HykuAddons::CsvParser"],
+           parser_klass: "HykuAddons::CsvParser",
+           export_source: work.class.to_s,
+           limit: 0)
+  end
+  let(:work) { create(:fully_described_work, user: depositor.email) }
+
+  before do
+    # Make sure default admin set exists
+    AdminSet.find_or_create_default_admin_set_id
+  end
+
+  after do
+    # TODO: cleanup export files and path
+  end
+
+  describe 'export job' do
+    before do
+      work
+    end
+
+    it 'creates csv and zip' do
+      Bulkrax::ExporterJob.perform_now(exporter.id)
+      expect(File.exist?(exporter.parser.setup_export_file)).to eq true
+      expect(File.exist?(exporter.exporter_export_zip_path)).to eq true
+      expect(File.readlines(exporter.parser.setup_export_file).size).to eq 2
+    end
+  end
+
+  describe 'single object export' do
+    subject(:parsed_metadata) { entry.parsed_metadata }
+    let(:entry) { exporter.entries.first }
+
+    before do
+      work
+      exporter.export
+    end
+
+    it 'populates all fields' do
+      expect(parsed_metadata['source_identifier']).to eq work.id
+      expect(parsed_metadata['model']).to eq 'GenericWork'
+      expect(parsed_metadata['depositor']).to eq depositor.email
+      expect(parsed_metadata['pagination']).to eq '1'
+      expect(parsed_metadata['event_date']).to eq '2009'
+      expect(parsed_metadata['related_exhibition_date']).to eq '2009'
+      expect(parsed_metadata['identifier'].split('|')).to contain_exactly('3-921099-34-X', 'doi:10.1038/nphys1170', 'ISBN:978-83-7659-303-6', '9790879392788', '3-540-49698-x', '0-19-852663-6', '978-3-540-49698-4')
```

Comment:
I would move the expected result values into their own variable to make this line smaller